### PR TITLE
Warn before losing unsubmitted post

### LINKED ISSUE #45: Warn before leaving a page with unsubmitted post
Please implement a [browser warning](https://developers.google.com/web/updates/2018/07/page-lifecycle-api#the-beforeunload-event) when leaving the page or closing the window, if there's unsubmitted text in the "write a new post", "quick post" or "new thread" fields.

Discussion: https://www.ivelt.com/forum/viewtopic.php?f=2&t=50578

### PR DESCRIPTION
**Pull request by nefshech**
Closes #45.

- When clicking on any link or closing the browser, a warning will be displayed if there's unsubmitted text.
- Unsubmitted text will be detected in the "new thread", "new post" or "quick post" forms.

### CODE REVIEW

File: extension/js/avoidLosingPost.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    window.addEventListener('beforeunload', avoidLosingPost);
+}
+
+function avoidLosingPost(event){
+    
+    var textarea = document.querySelector('textarea[name=message]');
+
+    if(textarea && textarea.value){
+        event.preventDefault();
+        return event.returnValue = 'Changes you made may not be saved. Are you sure you want to exit iVelt?';
```
Comment (kavhayushar) [inline extension/js/avoidLosingPost.js:17]: Message should be in Yiddish.

**Comment by nefshech** [comment]
It's only for the value to contain something. Newer browsers are not
showing custom messages anyway.

File: extension/js/avoidLosingPost.js (hunk 1 above)
Comment (nefshech) [reply extension/js/avoidLosingPost.js:17]: It's only for the value to contain something. Newer browsers are not
showing custom messages anyway.

**Comment by kavhayushar** [comment]
Since this is somewhat an intrusive behavior. I'm afraid we should first add an option to disable it before pushing it live.
Option should be in Setting Page to Enable/Disable this feature.

**Comment by nefshech** [comment]
> Since this is somewhat an intrusive behavior. I'm afraid we should first add an option to disable it before pushing it live. Option should be in Setting Page to Enable/Disable this feature.

Should this setting be released or only for testing purposes? I don't see a use case for this setting option (if tested properly)

**Comment by kavhayushar** [comment]
Of course released.
Every user should have the option if he wants to have this functionality.
We shouldn't force intrusive functionality on every user without the ability to opt-out.

I'm strongly for giving flexibility to users to turn on/off whichever feature they like/dislike.

**Comment by nefshech** [comment]
Makes sense, ok with me.

I was just comparing it to the keyboard shortcuts which we don't provide the option to opt out (even though some of them are overdone for some users) but I get it.

**Comment by kavhayushar** [comment]
@mordechairoth Should we wait for it, or make a run for it and add the settings later. So we don't miss the 1st update day?

**Comment by mordechairoth** [comment]
@kavhayushar  We should wait until there is a settings page.

**Comment by mordechairoth** [comment]
@nefshech the reason we don't want this to be turned on by default is because some users may get annoyed by this feature as @kavhayushar already mentioned

**Comment by nefshech** [comment]
@mordechairoth Understood.

I will rush it if it can still get into the next release, otherwise it will be done in the next few days. Please let me know

**Comment by mordechairoth** [comment]
@nefshech Don't rush, I will Iy"h put it into next months release.

**Comment by nefshech** [comment]
Added a setting, ON by default.

**Review by kavhayushar — APPROVED** [review]
Looks great and works well.
Thank you.

**Comment by bevekashaivelt** [comment]
@mordechairoth this should have been pushed in the December release. This and #51 as well, for some reason these were not merged before the release. I think it's unfair to wait another month, for these features to be released. I think another update should be pushed.

**Comment by kavhayushar** [comment]
@nefshech Are you avail to resolve the new conflicts?

**Comment by nefshech** [comment]
> @nefshech Are you avail to resolve the new conflicts?

Resolved.

**Pull request merged**